src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -266,6 +280,95 @@ def __init__(self, session: "snowflake.snowpark.session.Session") -> None:
         ] = None
         self._infer_schema_target_columns: Optional[List[str]] = None
 
+    def load(
+        self,
+        path: str,
+        format: Optional[
+            Literal["avro", "csv", "json", "xml", "parquet", "orc"]
+        ] = None,
+        schema: StructType = None,
+        **kwargs,
+    ) -> DataFrame:
+        """
+        Loads data from a file path.
+
+        This method is typically the last method in a daisy chain of DataFrame reader methods.
+        It allows specifying the file type using the `format` parameter, providing a schema using the `schema` parameter,
+        and passing any copy or format-specific options using the `kwargs` parameter.
+
+        Args:
+            path: The path to the file(s) to load.
+            format: The file format to use for ingestion. If not specified, it is assumed that it was already passed with an `option`.
+            schema: The schema to use for the loaded data.
+            **kwargs: Additional options to be passed for format-specific or copy-specific configurations.
+
+        Returns:
+            DataFrame: The loaded DataFrame.
+
+        Alternatively, you can use the `format` method and specify the format, and then use the `load` method to specify the path:
+            df = spark.read.format("csv").option("header", "true").load("path/to/your/file.csv")
+
+        The following example demonstrate how to use a DataFrameReader.
+            >>> # Create a temp stage to run the example code.
+            >>> _ = session.sql("CREATE or REPLACE temp STAGE mystage").collect()
+
+        Example 1:
+            Loading the first two columns of a CSV file and skipping the first header line:
+            >>> from snowflake.snowpark.types import StructType, StructField, IntegerType, StringType
+            >>> _ = session.sql("create file format if not exists csv_format type=csv skip_header=1 null_if='none'").collect()
+            >>> _ = session.file.put("tests/resources/testCSVspecialFormat.csv", "@mystage", auto_compress=False)
+            >>> # Define the schema for the data in the CSV files.
+            >>> schema = StructType([StructField("ID", IntegerType()),StructField("USERNAME", StringType()),StructField("FIRSTNAME", StringType()),StructField("LASTNAME", StringType())])
+            >>> # Create a DataFrame that is configured to load data from the CSV files in the stage with a given schema and format.
+            >>> df = session.read.load("@mystage/testCSVspecialFormat.csv",format="csv",schema=schema,format_name="csv_format")
+            >>> # Load the data into the DataFrame and return an array of rows containing the results.
+            >>> df.collect()
+            [Row(ID=0, USERNAME='admin', FIRSTNAME=None, LASTNAME=None), Row(ID=1, USERNAME='test_user', FIRSTNAME='test', LASTNAME='user')]
```

Comment:
![Screenshot 2023-07-05 at 10 32 43 AM](https://github.com/snowflakedb/snowpark-python/assets/107440467/c920b36a-4581-482d-a130-1be97942768d)

Please use [this](https://github.com/snowflakedb/snowpark-python/tree/main/docs) readme to generate docs locally and check if this was the intended result you want. You can dm me and I can help setup doc generation for you.

Reply:
Should be fine now